com.jetbrains.IntelliJ-IDEA-Community.json

```diff
@@ -4,22 +4,29 @@
     "runtime": "org.freedesktop.Sdk",
     "runtime-version": "20.08",
     "sdk": "org.freedesktop.Sdk",
+    "sdk-extensions": ["org.freedesktop.Sdk.Extension.openjdk11"],
     "separate-locales": false,
     "finish-args": [
         "--device=dri",
-        "--filesystem=host",
-        "--filesystem=xdg-run/keyring",
         "--share=ipc",
         "--share=network",
         "--socket=ssh-auth",
-        "--socket=wayland",
         "--socket=x11",
+	"--filesystem=~/IdeaProjects:create",
```

Comment:
Hope my comments are appreciated, so here goes:

Limiting the Flatpak to only a single folder, is way to limiting and will likely break 95% of all our user's workflow. For one, I have all my projects (Java and other) in ~/Projects and many others will likely have similar configurations.

It's also not really within the focus of Flathub to restrict the access of dev-tools so drastic. `--filesystem=home` would be a good compromise if you ask me, but no home access will be a hard sell. Really, until Jetbeans get involved, any sandboxing is better then no sandboxing at all.

Reply:
I understand this limitation. We could add a comment in the Flathub app description page to make aware of the change.
Another option, if we want to close this PR, could be to leave out the changes on file system access (maybe just changing it from host to home as you suggested) and create another PR limiting the file system access, removing the access to home. The new PR could be merged in a beta channel (I still don't know how to create it but I guess it could be possible).
Another limitation I can see is that users could have installed JDKs, Maven and other tools in other directories outside home, that they will not be able to access anymore without overriding file system permissions.

Reply:
Honestly, I don't find the restriction too annoying — except for the introduction part.

The limitation can be lifted by the user at any point by adding other location(s) — using `com.github.tchx84.Flatseal`, for instance. That's a one time action, because usually people use a predefined (set of) location(s)/path(s). Since IntelliJ IDEA auto-creates (and uses) by default `~/IdeaProjects`, it makes sense to use that location **by default**. Basically, this goes from being restrictive by default, and allowing the locations used by the IDE (and the plugins).

> Now comes to my mind that these are needed (from the ones I use): `~/.gradle`, `~/.java`, `~/.m2`, ~/.npm`, and `~/.gitconfig`.

In my opinion, this is the ideal way to go — but requires more work. Allowing `home` / `host` is the easiest way, but that approach wouldn't take advantage of what Flatpak offers.

In regard to the limitation with the tools installed outside user's home, that exists even if you allow `host`. I've only found (consistently between distros) that I can use any SDK in my home directory with (almost) no issues — hence I use `asdf` for every SDK. The point is, that limitation exists today with `host` access.

Reply:
Updated the filesystem permissions and adding filesystem access to ~/.java folder fixed the issue with the privacy policy agreement bug (updated description to reflect this).

Regarding the debate of home or only ~/IdeaProjects filesystem permission, it will be your call. I'm of course on the side of limiting unnecessary filesystem permissions.